docs/content/dagster-cloud/managing-deployments/controlling-logs.mdx

```diff
@@ -0,0 +1,96 @@
+---
+title: "Managing Compute Logs and Error Messages in Dagster Cloud | Dagster Docs"
+---
+
+# Managing compute logs and error messages in Dagster Cloud
+
+<Note>This guide is applicable to Dagster Cloud.</Note>
+
+In this guide, we'll cover how to adjust where Dagster Cloud compute logs are stored and manage masking of error messages in the Dagster Cloud UI.
+
+By default, Dagster Cloud ingests [structured event logs and compute logs](/concepts/logging#log-types) from runs and surfaces error messages from [code locations](/concepts/code-locations) in the UI.
+
+Depending on your organization's needs, you may want to retain these logs in your own infrastructure or mask error message contents.
+
+---
+
+## Modifying compute log storage
+
+Dagster's compute logs are handled by the configured [`ComputeLogManager`](/\_apidocs/internals#dagster.\_core.storage.compute_log_manager.ComputeLogManager). By default, Dagster Cloud utilizes the `CloudComputeLogManager` which stores logs in a Dagster Cloud-managed Amazon S3 bucket, but you can customize this behavior to store logs in a destination of your choice.
+
+### Writing to your own S3 bucket
+
+If using the Kubernetes agent, you can instead forward logs to your own S3 bucket by using the [`S3ComputeLogManager`](/\_apidocs/libraries/dagster-aws#dagster_aws.s3.S3ComputeLogManager).
+
+You can configure the `S3ComputeLogManager` in your [`dagster.yaml` file](/dagster-cloud/deployment/agents/customizing-configuration):
+
+```yaml
+compute_logs:
+  module: dagster_aws.s3.compute_log_manager
+  class: S3ComputeLogManager
+  config:
+    show_url_only: true
+    bucket: your-compute-log-storage-bucket
+    region: your-bucket-region
+```
+
+If you are using Helm to deploy the Kubernetes agent, you can provide the following configuration in your `values.yaml` file:
+
+```yaml
+computeLogs:
+  enabled: true
+  custom:
+    module: dagster_aws.s3.compute_log_manager
+    class: S3ComputeLogManager
+    config:
+      show_url_only: true
+      bucket: your-compute-log-storage-bucket
+      region: your-bucket-region
+```
+
+### Disabling compute log upload
+
+If your organization has its own logging solution which ingests `stdout` and `stderr` from your compute environment, you may want to disable compute log upload entirely. You can do this with the [`NoOpComputeLogManager`](/\_apidocs/internals#dagster.\_core.storage.noop_compute_log_manager.NoOpComputeLogManager).
+
+You can configure the `NoOpComputeLogManager` in your [`dagster.yaml` file](https://docs.dagster.io/dagster-cloud/deployment/agents/customizing-configuration):
+
+```yaml
+compute_logs:
+  module: dagster.core.storage.noop_compute_log_manager
+  class: NoOpComputeLogManager
+```
+
+If you are using Helm to deploy the Kubernetes agent, you may use the `enabled` flag to disable compute log upload:
```

Comment:
```suggestion
If you are using Helm to deploy the Kubernetes agent, use the `enabled` flag to disable compute log upload:
```

The `you may` makes this sound like a suggestion, when it's just how this needs to be done